games/ludii/jni_utils.h

```diff
@@ -30,11 +30,20 @@ SOFTWARE.
 
 #include <jni.h>  // NOLINT
 
+// TODO maybe we should optimise these checks out when
+// we're doing serious training runs/evaluations, since
+// we don't handle them other than printing them anyway?
+#define CHECK_JNI_EXCEPTION(jenv)                                              \
```

Comment:
Could this be refactored into a function? Let's try to keep macros to a minimum.
It could also throw a runtime_error as Java exceptions are likely unrecoverable?

Reply:
done